Fix ide hanging when file modified outside of it

maldoinc opened the pull request:
This PR fixes issue #112 

I could almost-reliably reproduce the issue at hand.

The problem would be that the message box is created as a modal window, then the main form is restored thus causing it (the message box) to stay behind the main form and causing it to freeze or simply not respond.

If you would drag away the form to reveal the message box then answer it the application starts responding normally.

This PR fixes this anomaly.

martijnlaan commented:
Thanks but I can't merge it like this until I fully understand why you changed it like this and why this fixes the problem, especially since this impacts everything and not just the IDE so I can't be sure it doesn't introduce new problems.

Also I wonder why the commit only changes the Unicode version. Does the non Unicode version not have this problem?

maldoinc commented:
The previous commit did indeed fix the issue but also broke a few things (namely message boxes from forms other than the main one).

To resolve this I reverted that one and added a check via the `IsIconic` function to detect wether the application is minimized and create a messagebox with a taskbar button.

The problem is mostly reproducible only when the application is minimized via the system minimize button.

martijnlaan commented:
Thanks!
